packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts

```diff
@@ -143,17 +143,20 @@ describeCompat("GC loading from older summaries", "NoCompat", (getTestObjectProv
 		summarizer1.close();
 		await reconnectSummarizerToBeElected(container2);
 
-		// Create a new alias data store so that the GC data changes without changing the GC state of existing data
-		// stores. This is to write the GC tree in summary (instead of handle) which is used for validation.
-		const ds2 = await containerRuntime.createDataStore(TestDataObjectType);
-		const aliasResult = await ds2.trySetAlias("root2");
-		assert.strictEqual(aliasResult, "Success", "Failed to alias data store");
+		// Call summarizeOnDemand. This will trigger the logic that processes summary ack. The ack from the last
+		// summary will cause summarizer2 to dispose because it sees a new ack that is generated by another summarizer.
 		await provider.ensureSynchronized();
-
-		// Summarize - summary3 with the new summarizer. Before it summarizes, it will catch up to latest and so the
-		// reference state of the data stores should be the same as in summary2.
-		// Also, note that while catching up, it will download summary2 and update state from it.
 		await summarizer2.summarizeOnDemand({ reason: "test" }).summarySubmitted;
+
+		// Wait for the container for the above summarizer to be disposed.
+		await timeoutPromise((resolve) => {
```

Comment:
This test needed update because previously, the ack from another summarizer was processed when the running summarizer started and before summarizeOnDemand was called. Now, the ack is processed async and there is no guarantee that it will be processed by the time summarizeOnDemand is called.

Reply:
All we have to do is make sure the ack is processed right? Isn't there a way to ensure we've processed s1's ack via just listening to the container/containerRuntime's ops?

Reply:
The problem is that just observing the ack isn't enough since the processing happens asynchronously. There has to be some trigger after the ack processing is complete and we don't have any.

Reply:
Feel free to resolve, I just ran into this issue myself